operators/src/adapters/raster_time.rs

```diff
@@ -128,14 +185,15 @@ where
 
                 if *current_spatial_tile + 1 >= num_spatial_tiles.expect("checked") {
                     // time slice ended => query next time slice of sources
-                    let mut next_qrect = *query_rect;
-                    next_qrect.time_interval = TimeInterval::new_unchecked(
+
+                    // advance current query rectangle
+                    query_rect.time_interval = TimeInterval::new_unchecked(
                         min(tile_a.time.end(), tile_b.time.end()),
                         query_rect.time_interval.end(),
                     );
-                    *time_end = None;
 
-                    stream.set(source_a(next_qrect).zip(source_b(next_qrect)));
+                    // reset the stream, the new stream will be created in the next iteration
+                    stream.set(None);
```

Comment:
Can we make it in such a way, that if the two streams are already temporally aligned, we do not query each time step again and again?

Reply:
this case would only arise if 
 - both time series are regular time series
 - both have the same start/reference and the same periodicity
 - there are nor gaps in the stream

for a given query processor output we don't know anything about the periodicity of its output. So unfortunately we can't avoid this. We would have to make additional metadata information about a queryprocessor output available.

Reply:
Well, it arises for our most prominent use case, the NDVI calculation from sentinel bands.

Why can't we just proceed if we query all the sources and everything matches?
I bet this really slows down everything.
Maybe we have to measure the overhead here.